### PR TITLE
Add WriteToV3IOStream.

### PR DESCRIPTION
**Pull request by gtopper**

### CODE REVIEW

File: storey/flow.py [hunk 1]
```diff
@@ -247,6 +250,16 @@ def __init__(self, webapi=None, access_key=None):
             'X-v3io-session-key': access_key
         }
 
+        self._put_records_headers = {
```
Comment (talIguaz) [inline storey/flow.py:253]: looks like all these headers are getting bigger and bigger, maybe move them to some other place like a v3io utility or something along with the build request/parse response

File: storey/flow.py [hunk 2]
```diff
@@ -370,6 +382,51 @@ def join_from_response(element, response):
         JoinWithHttp.__init__(self, request_builder, join_from_response, **kwargs)
 
 
+def _build_request_put_records(partition_func, events):
+    record_list_for_json = []
+    for event in events:
+        record = event.element
+        record_as_bytes = None
+        if isinstance(record, bytes):
```
Comment (talIguaz) [inline storey/flow.py:404]: what if the event is a dict?

File: storey/flow.py [hunk 3]
```diff
@@ -370,6 +382,51 @@ def join_from_response(element, response):
...
+        self._partition_func = partition_func
+
+        def request_builder(event):
+            request_body = _build_request_put_records(self._partition_func, [event])
+            return HttpRequest('POST', f'{self._webapi_url}/{stream_path}/', request_body, self._put_records_headers)
+
+        def join_from_response(_, response):
+            if response.status == 200:
+                response_obj = json.loads(response.body)
+                if response_obj['FailedRecordCount'] == 0:
```
Comment (talIguaz) [inline storey/flow.py:444]: whathappens if some records failed?

File: integration/test_flow_integration.py [hunk 4]
```diff
@@ -49,3 +59,18 @@ def test_join_with_http():
...
+    asyncio.run(SetupStream().setup(stream_path))
+    controller = build_flow([
+        Source(),
+        Map(lambda x: str(x)),
+        WriteToV3IOStream(stream_path, partition_func=lambda event: str(int(event.element) % 2))
+    ]).run()
+    for i in range(10):
+        controller.emit(i)
+
+    controller.terminate()
```
Comment (talIguaz) [inline integration/test_flow_integration.py:130]: actually assert that values were written, using v3io-py/nginx or whatever

File: storey/flow.py (hunk 2 above)
Comment (gtopper) [reply storey/flow.py:404]: You need to serialize it first? :)
I guess I could add json.dumps in that case, to be nice.

File: storey/flow.py (hunk 1 above)
Comment (gtopper) [reply storey/flow.py:253]: Maybe...

File: storey/flow.py (hunk 3 above)
Comment (gtopper) [reply storey/flow.py:444]: `raise V3ioError(f'Failed to put records to V3IO. Got {response.status} response: {response.body}')`

File: integration/test_flow_integration.py (hunk 4 above)
Comment (gtopper) [reply integration/test_flow_integration.py:130]: Yeah... Not excited about that.

File: storey/flow.py (hunk 1 above)
Comment (gtopper) [reply storey/flow.py:253]: Actually it depends on the access key, so not really convenient to move it. It won't grow much bigger anyway :)

File: storey/flow.py [hunk 5]
```diff
@@ -382,11 +397,11 @@ def join_from_response(element, response):
         JoinWithHttp.__init__(self, request_builder, join_from_response, **kwargs)
 
 
-def _build_request_put_records(partition_func, events):
+def _build_request_put_records(shard_id, events):
+    print(f'_build_request_put_records({shard_id}, {events})')
```
Comment (talIguaz) [inline storey/flow.py]: remove print

File: storey/flow.py [hunk 6]
```diff
@@ -396,42 +411,128 @@ def _build_request_put_records(partition_func, events):
...
+        Flow.__init__(self, **kwargs)
         NeedsV3ioAccess.__init__(self, webapi, access_key)
-        if partition_func is None:
-            def f(_):
-                return str(random.random())
 
-            partition_func = f
-        self._partition_func = partition_func
+        self.stream_path = stream_path
+        self._sharding_func = sharding_func
```
Comment (talIguaz) [inline storey/flow.py:431]: validate callable?

**Review by talIguaz — APPROVED** [review]

**Pull request merged**